app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
Inetersting thoughs, However keep in mind, that the plan needs to happen *before* the execution starts (that's the whole point of the planning mechanism). If you want to do this kind of decisions, we would need a new concept and I would strictly not recommend trying to add that in to the planning phase. If there is some need for advanced processing of the output, it would need to happen in the `run` phase and I can imaging helpers/concepts to make it easier. But by definition, it can't be done i plan phase.